src/adt/print/tree.jl

```diff
@@ -136,20 +135,118 @@ function (p::Printer)(node)
     end
 end
 
+function children(node)
+    error("unimplemented children method for $(typeof(node))")
+end
 function print_node(io::IO, node)
     error("unimplemented print_node method for $(typeof(node))")
+end    
+function print_child_annotation(io::IO, node, child, annotation)
```

Comment:
do we actually need a custom printing of the annotation/keys? they are usually directly printed anyway

Reply:
Yes, it is absolutely needed. For example, in YaoExpr, we want to print coeffecients for each term in an OpTerm `Add` using an inline printer for scalars. We can not achieve this without custom printing.